core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -630,7 +634,10 @@ class SparkContext(config: SparkConf) extends SparkStatusAPI with Logging {
    * necessary info (e.g. file name for a filesystem-based dataset, table name for HyperTable),
    * using the older MapReduce API (`org.apache.hadoop.mapred`).
    *
-   * @param conf JobConf for setting up the dataset
+   * @param conf JobConf for setting up the dataset. Note: This will be put into a Broadcast.
```

Comment:
Nice find. It seems perfectly reasonable from the user's perspective to just save `sc.hadoopConfiguration` into a val and use it for many things. That's probably what I would have done if I didn't know about the nuances here.